### DISCUSSION TITLE
Whenever I use a biome that isn't Biome.PLAINS nothing will generate

### DISCUSSION THREAD

**Discussion by Flamgop** [Minestom/Minestom#681] [Help] [answered]
The biome just says "Waiting for chunk..."

**Comment by TheMode** [comment, marked as answer]
We need a reproducer.

**Reply by Flamgop** [reply]
Ok,
1. Make a chunk populator that sets the biome to anything other than default plains biome
2. add said chunk populator to the populators list
3. run server

my code looks like this
```java
for (int x = 0; x < 16; x++) {
  for (int z = 0; z < 16; z++) {
    for (int y = 0; y < 256; y++) {
        float humidity = Math.abs(humidityNoise.GetNoise(x, y, z));
        float temperature = Math.abs(temperatureNoise.GetNoise(x, y, z));
  
        if (isDesert(temperature, humidity)) chunk.setBiome(x, y, z, DESERT);
        else if (isForest(temperature, humidity)) chunk.setBiome(x, y, z, FOREST);
        else if (isPlains(temperature, humidity)) chunk.setBiome(x, y, z, Biome.PLAINS);
        else if (isMountain(temperature, humidity)) chunk.setBiome(x, y, z, MOUNTAINS);
        else if (isSnow(temperature, humidity)) chunk.setBiome(x, y, z, SNOW);
        else if (isJungle(temperature, humidity)) chunk.setBiome(x, y, z, JUNGLE);
        else chunk.setBiome(x, y, z, FOREST);
    }
  }
}
```
(this is in the populateChunk method, the `isBiome` methods just compare humidity and temperature, and even with the else statement at the end it still fails to set it correctly, All of these biomes are passed as parameters to the MinecraftServer.getBiomeManager().addBiome() method)

**Reply by TheMode** [reply]
Reproducer must include a complete (and minimal) server. We are otherwise just guessing.

**Reply by Flamgop** [reply]
[reproducer.zip](https://github.com/Minestom/Minestom/files/8068604/reproducer.zip)
Here ya go

**Reply by Flamgop** [reply]
[minimized-reproducer.zip](https://github.com/Minestom/Minestom/files/8073924/minimized-reproducer.zip)
heres it with the minimal amount of code that should get this issue to occur

**Reply by Flamgop** [reply]
wait nvm i just found the issue and its stupid lol
